### DISCUSSION TITLE
Default ConfigureAwait in extension method and VSTHRD003 warning

### DISCUSSION THREAD

**Discussion by alexreadmail** [microsoft/vs-threading#1064] [Q&A] [answered]
I created an extension for Task to introduce the DefaultAwait method.
But I see VSTHRD003  warning.
Is it safe and could it be suppressed?

```cs
internal static class TaskExtensions
{
    private const bool DefaultAwaitResponse = false;

    public static ConfiguredTaskAwaitable DefaultAwait(this Task task)
    {
        Guard.IsNotNull(task);

        return task.ConfigureAwait(DefaultAwaitResponse);
    }

    public static ConfiguredTaskAwaitable<T> DefaultAwait<T>(this Task<T> task)
    {
        Guard.IsNotNull(task);

        return task.ConfigureAwait(DefaultAwaitResponse);
    }
}
```

**Comment by AArnott** [comment, marked as answer]
If VSTHRD003 is showing up in your `TaskExtensions` class itself, yes, you can suppress it. 
What I would be more concerned about is whether code that *uses* `await task.DefaultAwait()` would *not* produce VSTHRD003 when it should. You might want to test that.

**Reply by alexreadmail** [reply]
Thank you. We have a rule in .editorconfig and shows a warning if you don't use task.DefaultAwait() or task.ConfigureAwait(false or true).
